src/google/protobuf/compiler/objectivec/objectivec_helpers.h

```diff
@@ -167,7 +167,15 @@ string GetOptionalDeprecatedAttribute(
     isDeprecated = file->options().deprecated();
```

Comment:
If you really want to point folks at the right thing, you could also track when the marking was found here (a file) vs. on a field, and adjust your messaging.

Reply:
Good idea! Putting together the changes to make that happen.

Reply:
Completed and verified 👌